events-api/src/modules/events/events.controller.ts

```diff
@@ -23,3 +29,64 @@ EventsController.get(
     });
   }
 );
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsCreateDtoSchema),
+  async (req, res) => {
+    await EventsService.createEvent(req.body);
+
+    return res.status(201).json({
+      message: "Event created successfully",
+    });
+  }
+);
+
+EventsController.patch(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  validateRequestBody(eventsUpdateDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.updateEvent(
+      req.body,
+      req.params["id"] as string
+    );
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(201).json({
+      message: "Event updated successfully",
+    });
+  }
+);
+
+EventsController.delete(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.deleteEvent(req.params["id"] as string);
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(204).json({
+      message: "Event deleted successfully",
```

Comment:
This endpoint should also return the event data: https://github.com/NegatiV-64/expressjs-masterclass-2024/blob/main/homework/README.md?plain=1#L71

Reply:
![image](https://github.com/user-attachments/assets/6ef9fe3a-64e9-47ce-b768-f05e434d4220)

Reply:
But what stops you from:

- Selecting an existing record
- Deleting a record?

Reply:
done, but I have changed status from 204 => 200 since postman not showing response content 

![image](https://github.com/user-attachments/assets/ebf2217e-9cb5-4295-86cb-50b70d0b17fa)